modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/AnnotationProcessorUtils.java

```diff
@@ -84,24 +92,48 @@ private AnnotationProcessorUtils()
     }
 
     /**
-     * Scans all classes in the {@code roundEnvironment} looking
-     * for methods annotated with {@link Operation}.
+     * Scans all the classes annotated with {@link Extension}, takes
+     * the {@link Operations} from those and returns the methods
+     * which are public and are not annotated with {@link Ignore}
      *
      * @param roundEnvironment the current {@link RoundEnvironment}
      * @return a {@link Map} which keys are the method names and the values are the
      * method represented as a {@link ExecutableElement}
      */
-    static Map<String, ExecutableElement> getOperationMethods(RoundEnvironment roundEnvironment)
+    static Map<String, Element> getOperationMethods(RoundEnvironment roundEnvironment, ProcessingEnvironment processingEnvironment)
     {
-        ImmutableMap.Builder<String, ExecutableElement> methods = ImmutableMap.builder();
-        for (Element rootElement : roundEnvironment.getRootElements())
+        ImmutableMap.Builder<String, Element> methods = ImmutableMap.builder();
+        for (Element rootElement : roundEnvironment.getElementsAnnotatedWith(Extension.class))
         {
             if (!(rootElement instanceof TypeElement))
             {
                 continue;
             }
 
-            methods.putAll(getMethodsAnnotatedWith((TypeElement) rootElement, Operation.class));
+            Class<?> extension = classFor((TypeElement) rootElement, processingEnvironment);
+            Class<?>[] operationsClasses = extension.getAnnotation(Operations.class).value();
+
+            List<AnnotationValue> annotationValues = IntrospectionUtils.getAnnotationFieldValue(rootElement, Operations.class, VALUE);
+            for (Class<?> operationClass : operationsClasses)
+            {
+                Element operationClassElement = annotationValues.stream()
+                        .map(e -> ((DeclaredType) e.getValue()).asElement())
+                        .filter(e -> e.getSimpleName().toString().equals(operationClass.getSimpleName())).findFirst().get();
+
+                Collection<Method> operations = IntrospectionUtils.getOperationMethods(operationClass);
+                for (Method operation : operations)
+                {
+                    Element operationMethodElement = operationClassElement.getEnclosedElements().stream()
+                            .filter(e -> e.getSimpleName().toString().equals(operation.getName()))
+                            .findFirst().orElse(null);
+
+                    if (operationMethodElement != null)
```

Comment:
since findFirst returns an Optional, either use "ifPresent" to do the put, or as "isPresent" in the if clause. This avoids the != null comparison.